samza-core/src/main/java/org/apache/samza/startpoint/StartpointObjectMapper.java

```diff
@@ -39,12 +40,16 @@ static ObjectMapper getObjectMapper() {
     SimpleModule module = new SimpleModule("StartpointModule", new Version(1, 0, 0, ""));
     module.addSerializer(Instant.class, new CustomInstantSerializer());
     module.addDeserializer(Instant.class, new CustomInstantDeserializer());
-
     objectMapper.registerModule(module);
-    objectMapper.registerSubtypes(StartpointSpecific.class);
-    objectMapper.registerSubtypes(StartpointTimestamp.class);
-    objectMapper.registerSubtypes(StartpointUpcoming.class);
-    objectMapper.registerSubtypes(StartpointOldest.class);
+
+    // 1. To support polymorphism for serialization, the Startpoint subtypes must be registered here.
+    // 2. The NamedType container class provides a logical name as an external identifier so that the full canonical
```

Comment:
Just double checking: Does the "logical name" allow for easily adding new types in the future? Is there a certain way new types need to be added? For example, if the "logical name" is just a counter, then we would need to add a new subtype at the end of the list.

Reply:
Not necessarily easier, but it seems to be a best practice based on the examples I've seen. Underneath the hood, it just uses the simple class name and not the full canonical name, so ordering doesn't matter. The `NamedType` container constructor also has a 2nd optional parameter where we can specify the logical name ourselves too.

Reply:
Sounds good. Just wanted to make sure there wasn't any gotcha with using the new way. I didn't need it to be easier than the old way. It's good that it makes the serialization more compact.